provider_shopper/lib/main.dart

```diff
@@ -30,7 +30,7 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProxyProvider<CatalogModel, CartModel>(
           create: (context) => CartModel(),
           update: (context, catalog, cart) {
-            cart.catalog = catalog;
+            cart!.catalog = catalog;
```

Comment:
This seems unsafe. How do we _know_ we can assert non-null here when the `Provider` itself can't guarantee it? See `update`'s signature [here](https://pub.dev/documentation/provider/5.0.0-nullsafety.5/provider/ProxyProviderBuilder.html).

Reply:
Exactly, I think instead of `!` operator `?.` can be used ?

Reply:
You'll need to try. Maybe it'll make more sense to have a guard clause, such as `if (cart == null) return;`.